api/capi/src/nnstreamer-capi-single.c

```diff
@@ -343,19 +343,25 @@ ml_single_set_info_in_handle (ml_single_h single, gboolean is_input,
     configured = klass->output_configured (single_h->filter);
   }
 
-  if (tensors_info) {
-    if (!configured)
-      ml_single_set_inout_tensors_info (filter_obj, is_input, tensors_info);
-    ml_tensors_info_clone (dest, tensors_info);
-  } else {
-    ml_tensors_info_h info;
-
+  if (configured) {
+    /* get configured info and compare with input info */
     ml_single_get_tensors_info_from_filter (single_h->filter, is_input, &info);
+
+    if (tensors_info && !ml_tensors_info_is_equal (tensors_info, info)) {
+      /* given input info is not matched with configured */
+      goto done;
```

Comment:
Is this ok to return TRUE with this? (it appears to be able to do so..)

Reply:
I added simple testcase with ml_tensors_info_is_equal and ml_tensors_info_is_valid

Reply:
Ok.. I mean is it ok to return TRUE when ml_tensor_info_is_equal is FALSE.

Reply:
I added flag is_valid which validates the tensor-info at the last step.